api/api-config-repos-v2/src/test/groovy/com/thoughtworks/go/apiv2/configrepos/ConfigReposControllerV2Test.groovy

```diff
@@ -72,6 +73,74 @@ class ConfigReposControllerV2Test implements SecurityServiceTrait, ControllerTra
     new ConfigReposControllerV2(new ApiAuthenticationHelper(securityService, goConfigService), service, entityHashingService)
   }
 
+  @Nested
+  class DeprecationHeaders {
+    @Nested
+    class Index implements DeprecationHeadersTrait {
+      @Override
+      String getControllerMethodUnderTest() {
+        return "index"
+      }
+
+      @Override
+      void makeHttpCall() {
+        getWithApiHeader(controller.controllerBasePath(), [:])
+      }
+    }
```

Comment:
`DeprecationHeadersTrait` verifies whether appropriate deprecation headers are added or not, based on whether the controller is deprecated.

Ideally, this test should be added for all the controllers to verify these headers are present or not, based on whether the controller is marked as deprecated.

Is there a way to run a verification against all the registered route of a controller?
\cc: @maheshp @ketan @bdpiparva 

Note: Its possible through reflection, but the test becomes messy and unreadable :(.